### PR TITLE
Bug 1566924 - renaming CRD resources

### PR DESCRIPTION
**Pull request by shawn-hurley**
Updating the name of custom resource definition types.

### CODE REVIEW

**Comment by shawn-hurley** [comment]
@eriknelson @jmrodri 

I think I still need to update the artifacts before this can be merged, but worth looking at now I think. 

note that everything in clients package is generated.

File: pkg/apis/automationbroker/v1alpha1/types.go [hunk 1]
```diff
@@ -14,12 +14,17 @@ See the License for the specific language governing permissions and
...
 
-package v1
+package v1alpha1
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// LocalObjectReference - reference to an object in the same namespace.
+type LocalObjectReference struct {
```
Comment (eriknelson) [inline pkg/apis/automationbroker/v1alpha1/types.go:24]: :+1:

File: pkg/apis/automationbroker/v1alpha1/types.go [hunk 2]
```diff
@@ -71,25 +77,41 @@ type BundleSpec struct {
 	Plans []Plan `json:"plans"`
 }
 
+// Status - The status for the bundle
+type Status string
+
+const (
+	// ErrorStatus - Bundle error status
+	ErrorStatus Status = "error"
+	// OKStatus - Bundle oke status
```
Comment (eriknelson) [inline pkg/apis/automationbroker/v1alpha1/types.go]: `oke` -> `ok`

File: pkg/apis/automationbroker/v1alpha1/types.go [hunk 3]
```diff
@@ -71,25 +77,41 @@ type BundleSpec struct {
...
-	Free           bool         `json:"free"`
-	Bindable       bool         `json:"bindable"`
-	UpdatesTo      []string     `json:"updates_to,omitempty"`
-	Parameters     []Parameters `json:"parameters,omitempty"`
-	BindParameters []Parameters `json:"bindParameters,omitempty"`
+	Metadata       string      `json:"metadata"`
+	Free           bool        `json:"free"`
+	Bindable       bool        `json:"bindable"`
+	UpdatesTo      []string    `json:"updates_to,omitempty"`
+	Parameters     []Parameter `json:"parameters,omitempty"`
```
Comment (eriknelson) [inline pkg/apis/automationbroker/v1alpha1/types.go:106]: :+1:

File: pkg/apis/automationbroker/v1alpha1/types.go [hunk 4]
```diff
@@ -161,79 +161,95 @@ const (
...
 	// Store the parameters as a json encoded string.
 	Parameters string `json:"parameters"`
-	JobToken   string `json:"jobToken"`
+}
+
+// BundleBindingStatus - status of the bundle
+type BundleBindingStatus struct {
+	State           State          `json:"state"`
+	LastDescription string         `json:"lastDescription,omitempty"`
+	Jobs            map[string]Job `json:"jobs"`
```
Comment (eriknelson) [inline pkg/apis/automationbroker/v1alpha1/types.go:208]: So if I want the latest Job status, I can find it as the last item in this slice?

**Review by eriknelson — APPROVED** [review]
LGTM, just one minor comment misspelling.

File: pkg/apis/automationbroker/v1alpha1/types.go (hunk 4 above)
Comment (shawn-hurley) [reply pkg/apis/automationbroker/v1alpha1/types.go:208]: it's a map, so you would need the token. 

If you didn't know the token, but wanted the last action then you would have to loop through and compare the last modified timestamp.

Comment (eriknelson) [reply pkg/apis/automationbroker/v1alpha1/types.go:208]: Derp, thank you. That's what I get for skimming this.

**Review by jmrodri — APPROVED** [review]
ACK, not a fan of the copyright format, but that's a nitpick. Approving.

**Pull request merged**